### ISSUE TITLE
Rpm is unsyncronized with animations

### ISSUE THREAD

**Issue by The-idk-notdot** [MCModderAnchor/TACZ#139]
Guns are working correctly only if there is less than 20 ms ticks, if the value is higher,the ammo counter depletes way slower, no matter is there 20 tps or not.

**Comment by The-idk-notdot** [comment]
https://github.com/user-attachments/assets/eaeca01d-afe3-4049-8ea0-888cf5ac8de9


https://github.com/user-attachments/assets/f74112f2-b323-408f-90aa-91af4b8d167b

**Comment by The-idk-notdot** [comment]
> slow.mp4 
>  fast.mp4

The bug makes mod literally unplayable, if the citadel's chunk gen spawn modifier is higher than 1.0

**Comment by 286799714** [comment]
Sorry, I didn't get the point from the video. We use timestamp to mark rpm timeout, which is different from just counting the ticks. I think this may be the reason

**Comment by The-idk-notdot** [comment]
> Sorry, I didn't get the point from the video. We use timestamp to mark rpm timeout, which is different from just counting the ticks. I think this may be the reason

In the first video rpm is slower than in the second.

**Comment by The-idk-notdot** [comment]
While main game runs normally, guns are shooting slower than expected.
As i said, it greatly affects large modpacks.

**Comment by The-idk-notdot** [comment]
@286799714

**Comment by The-idk-notdot** [comment]
Will it be fixed or not

**Comment by 286799714** [comment]
> Will it be fixed or not

Sorry for the late reply. We are currently fixing other issues and this bug needs to be archived for now. According to your feedback, tacz's rpm system does not work properly when used with the "citadel's chunk gen" mod, is that correct?

**Comment by The-idk-notdot** [comment]
> > Will it be fixed or not
> 
> Sorry for the late reply. We are currently fixing other issues and this bug needs to be archived for now. According to your feedback, tacz's rpm system does not work properly when used with the "citadel's chunk gen" mod, is that correct?

No, the more ms delay the slower the weapon fires, but the game itself works as usual. Citadel just spawns extra mobs, which is adding 5-7 ms.

**Comment by The-idk-notdot** [comment]
I tried to delete citadel, but nothing changed. Looks like its not a mod incompatibility, because i launched tacz on another modpack with diffrent mods but same count of them, and guns shot slower, like in the first modpack

**Comment by The-idk-notdot** [comment]
Also, here is demonstration video with cz75:
In first video, rpm is slow, in second fast.

https://github.com/user-attachments/assets/2562c1b8-9282-43ba-9c57-5fafcc137432


https://github.com/user-attachments/assets/13be0314-3b3c-4f6d-b1bc-c27cf72e3cf9

**Comment by The-idk-notdot** [comment]
Also, old t

https://github.com/user-attachments/assets/eb240510-2755-4fbf-8e76-69bf2160ed21

ac didnt't have this problem. Did it use the same rpm system as tacz?
